### PR TITLE
mon: drop QuorumService and cleanups.

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by neha-ojha** [comment]
@tchaikov can you please rebase?

**Comment by tchaikov** [comment]
@neha-ojha rebased.

**Comment by tchaikov** [comment]
@jdurgin @neha-ojha ping?

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-12-03_06:25:29-rados-wip-kefu-testing-2020-12-02-2319-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-12-04_02:12:52-rados-wip-kefu-testing-2020-12-02-2319-distro-basic-smithi/

**Comment by tchaikov** [comment]
@jecluis ping?

**Comment by tchaikov** [comment]
rebased against master again.

**Review by jecluis — APPROVED** [review]

File: src/mon/Monitor.h [hunk 1]
```diff
@@ -871,13 +871,13 @@ class Monitor : public Dispatcher,
         }
         ss << "cmd='" << m->cmd << "': finished";
 
-        mon->audit_clog->info() << ss.str();
-	mon->reply_command(op, rc, rs, rdata, version);
+        mon.audit_clog->info() << ss.str();
+	mon.reply_command(op, rc, rs, rdata, version);
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/mon/Monitor.h:875]: it's just tab versus spaces. the original code uses both of them...  and github do not render tab as 8 spaces. neither do our browsers. will use spaces anyway.

File: src/mon/MonmapMonitor.cc [hunk 2]
```diff
@@ -1086,20 +1086,20 @@ bool MonmapMonitor::prepare_command(MonOpRequestRef op)
...
       //okay, initial arguments make sense, check pools and cluster state
-      err = mon->osdmon()->check_cluster_features(CEPH_FEATUREMASK_STRETCH_MODE, ss);
+      err = mon.osdmon()->check_cluster_features(CEPH_FEATUREMASK_STRETCH_MODE, ss);
       if (err)
 	goto reply;
       struct Plugger {
-	Paxos *p;
-	Plugger(Paxos *p) : p(p) { p->plug(); }
-	~Plugger() { p->unplug(); }
+	Paxos &p;
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/mon/MonmapMonitor.cc:1093]: this source file is full of tabs. so i will leave as it is.

File: src/mon/Monitor.h [hunk 3]
```diff
@@ -871,13 +871,13 @@ class Monitor : public Dispatcher,
...
       }
       else if (r == -ECANCELED)
         return;
       else if (r == -EAGAIN)
-	mon->dispatch_op(op);
+	mon.dispatch_op(op);
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/mon/Monitor.h]: indeed..

**Pull request merged**